test/Microsoft.DotNet.DarcLib.Tests/VirtualMonoRepo/VmrRegistrationsTests.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using FluentAssertions;
+using Microsoft.DotNet.DarcLib.VirtualMonoRepo;
+using Microsoft.DotNet.Internal.DependencyInjection.Testing;
+using NUnit.Framework;
+
+namespace Microsoft.DotNet.DarcLib.Tests.VirtualMonoRepo;
+
+public class VmrRegistrationsTests
+{
+    /// <summary>
+    /// Tests registering of dependencies for GitNativeRepoCloner
+    /// </summary>
+    [Test]
+    public void AddGitNativeRepoClonerSupportRegistrationIsCoherent()
+    {
+        DependencyInjectionValidation.IsDependencyResolutionCoherent(sc => VmrRegistrations.AddGitNativeRepoClonerSupport(sc),
```

Comment:
FYI this could be used like this as it's an extension method

```suggestion
        DependencyInjectionValidation.IsDependencyResolutionCoherent(sc => sc.AddGitNativeRepoClonerSupport(),
```